### PR TITLE
made the remaining configuration props configurable by config file

### PR DESCRIPTION
**Pull request by pwinckles**
**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3627

# What does this Pull Request do?

Moves all of the remaining config props to Spring config

# How should this be tested?

Try running Fedora by specifying these props in a config file and Fedora should pickup the values.

# Interested parties
@fcrepo/committers

### CODE REVIEW

**Comment by pwinckles** [comment]
Example config:

```
fcrepo.auth.webac.groupAgent.baseUri=http://user.example.com/
fcrepo.auth.webac.userAgent.baseUri=http://group.example.com/
fcrepo.auth.webac.authorization=/home/winckles/workspace/fcrepo/fcrepo4/fcrepo-auth-webac/src/test/resources/test-root-authorization2.ttl
fcrepo.session.timeout=1000
fcrepo.jms.baseUrl=http://jms.example.com
fcrepo.properties.management=relaxed
fcrepo.velocity.runtime.log=/var/tmp/velocity.log
```

File: fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/session/TransactionProviderTest.java [hunk 1]
```diff
@@ -34,14 +34,13 @@
...
 import static java.net.URI.create;
-import static org.fcrepo.http.commons.session.TransactionProvider.JMS_BASEURL_PROP;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 /**
- * @author pwinckles
+ * @author pwinckles (I definitely did not write these)
```
Comment (whikloj) [inline fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/session/TransactionProviderTest.java:43]: I'm not a huge fan of this as makes it seem like there is something wrong with the tests.

Based on the PR you added it with (https://github.com/fcrepo/fcrepo/pull/1661) you say you moved it here, but I can't see where you moved it from. If we can determine that, then we can identify the actual author, though now that you have altered it you'd be an author as well.

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 2]
```diff
@@ -316,7 +326,7 @@ private static String stripUserAgentBaseURI(final String object) {
      *                    resource
      * @return a list of acl:Authorization objects
      */
-    private static List<WebACAuthorization> getAuthorizations(final FedoraResource aclResource,
+    private List<WebACAuthorization> getAuthorizations(final FedoraResource aclResource,
                                                               final boolean ancestorAcl) {
 
         final List<WebACAuthorization> authorizations = new ArrayList<>();
```
Comment (whikloj) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java:332]: Can you also remove the two lines *below* this one.
```java
//TODO figure out where the translator should be coming from
final IdentifierConverter<Resource, FedoraResource> translator = null;
```
I don't think they are needed anymore

**Review by whikloj — CHANGES_REQUESTED** [review]
A couple things, but otherwise this is nice. I like how easy it makes it to perform testing.

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 3]
```diff
@@ -257,7 +268,7 @@
     /**
      * Given a FedoraResource, return a list of agents.
      */
-    private static Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
+    private Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
```
[the remark this answers is not in the document]
Comment (pwinckles) [reply fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java:271]: I removed it in the commits I have together for the `fcrepo-config.xml` PR. Is it okay to wait for that PR to remove?

File: fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/session/TransactionProviderTest.java (hunk 1 above)
Comment (pwinckles) [reply fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/session/TransactionProviderTest.java]: Removed

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java (hunk 2 above)
Comment (pwinckles) [reply fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java:332]: I removed it in the commits I have together for the `fcrepo-config.xml` PR. Is it okay to wait for that PR to remove?

**Comment by pwinckles** [comment]
Rebased to pull in the fix for the failing itest

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**